adr/ADR-33.md

```diff
@@ -0,0 +1,236 @@
+# Service API
+
+| Metadata | Value                 |
+| -------- | --------------------- |
+| Date     | 2022-11-23            |
+| Author   | @aricart              |
+| Status   | Partially Implemented |
+| Tags     | client                |
+
+## Context and Problem Statement
+
+Simplify the development of NATS micro-services.
+
+The design goal of the API is to reduce development to having a complexity
+similar to that of writing a NATS subscription, but then by using simple
+configuration, allow the specified metadata to allow for standardization of
+discovery and observability.
+
+## Design
+
+Service configuration relies on the following:
+
+- `name` - really the _kind_ of the service. Shared by all the services that
+  have the same name. This `name` cannot have any spaces or periods.
+- `version` - an opaque string identifying the version
+- `description` - a human-readable description about the service
+- `schema`:
+  - `request` - a string/url describing the format of the request payload
```

Comment:
using json schema or what?

Reply:
currently not making any assumptions on what that payload is. So it could be an URL or inlined.